Background - update random forest description (#184)
chhoumann/thesis-chemcam: report_thesis/src/sections/background/ensemble_learning_models/etr.tex

\gls{rf} is an ensemble learning method that improves the accuracy and robustness of decision trees by building multiple trees and combining their predictions.
Each tree is trained on a random subset of the data using bootstrap sampling, where samples are drawn with replacement, meaning the same sample can be selected multiple times.

> Bagging, or bootstrap aggregating, uses these bootstrap samples to create multiple versions of the training set.

Reviewer: Need to clarify what those subsets are actually used for
```suggestion
Bagging, or bootstrap aggregating, involves training each tree on a different bootstrap sample and then aggregating their predictions to form the final output.
```

> This introduces variability and reduces overfitting, as some data points may appear multiple times while others may be omitted.

Reviewer: This introduces variability ... in the (training data available to the) models?

Could clarify that, since we later pose reduced variance as a desirable trait (for reducing overfitting)

> This introduces variability and reduces overfitting, as some data points may appear multiple times while others may be omitted.

Author: ```suggestion
This introduces variability in the training data available to the models and reduces overfitting, as some data points may appear multiple times while others may be omitted
```

In addition to bootstrap sampling, random forests introduce an additional layer of randomness by selecting a random subset of features for splitting at each node of the trees.
This further decorrelates the trees, enhancing the model's robustness and reducing the risk of overfitting.
By aggregating predictions from multiple trees, the model achieves better generalization and robustness.

For a feature vector $x$, the prediction of a \gls{rf} model can be represeted as an aggregation of the predictions of individual trees:

$$
f(x) = \frac{1}{M} \sum_{m=1}^{M} f_m(x),
$$

where $f_m(x)$ is the prediction of the $m$-th tree, and $M$ is the total number of trees.

> This aggregation is a form of bagging, which averages the predictions of the individual trees to produce a final prediction, reducing the overall variance of the model.

Reviewer 2: I feel like saying this is a form of bagging might be confusing to the reader. While true, it sort of blurs the lines in an otherwise clear explanation.

> This aggregation is a form of bagging, which averages the predictions of the individual trees to produce a final prediction, reducing the overall variance of the model.

Reviewer 2: ```suggestion
 This form of aggregation reduces the overall variance of the model, by averaging the predictions of the individual trees to produce a final prediction.
```

> This aggregation is a form of bagging, which averages the predictions of the individual trees to produce a final prediction, reducing the overall variance of the model.

Reviewer 2: ```suggestion
This form of aggregation reduces the overall variance of the model, by averaging the predictions of the individual trees to produce a final prediction.
```

Reviewer: I think @Ivikhostrup's is good 👍
